test/ExecutionTests.jl

```diff
@@ -216,10 +216,14 @@ tune!(b)
     "good")
 @test_throws UndefVarError local_var
 @benchmark some_var = "whatever" teardown = (@test_throws UndefVarError some_var)
-@benchmark foo, bar = "good", "good" setup = (foo = "bad"; bar = "bad") teardown = (@test foo ==
-                                                                                          "good" &&
-    bar ==
-                                                                                          "good")
+@benchmark foo, bar = "good", "good" setup = (foo = "bad"; bar = "bad") teardown = @test(
+    foo == "good" && bar == "good"
+)
+if VERSION >= v"1.7"
+    @benchmark (; foo, bar) = (foo="good", bar="good") setup = (foo = "bad"; bar = "bad") teardown = @test(
+        foo == "good" && bar == "good"
+    )
+end
```

Comment:
These tests are good, but if I understand correctly, I think this is missing the case where we interpolate a value from local scope with dollar, since that's part of the code that this PR changes.

Reply:
I've added tests for interpolated values both for the tuple and named tuple case (though I don't think the code I changed is specific to interpolation in any way).